cpp/arcticdb/version/version_core.cpp

```diff
@@ -281,6 +290,8 @@ VersionedItem update_impl(
 
     auto index_segment_reader = index::get_index_reader(*(update_info.previous_index_key_), store);
     util::check_rte(!index_segment_reader.is_pickled(), "Cannot update pickled data");
+    sorting::check<ErrorCode::E_UNSORTED_DATA>(frame.desc.get_sorted() == arcticc::pb2::descriptors_pb2::SortedValue::ASCENDING && std::holds_alternative<stream::TimeseriesIndex>(frame.index),
```

Comment:
The `TimeseriesIndex` is already checked on line 298 below. You might want to move that check up.

Reply:
I disagree here. We do that check much later and only under a specific type of the row filter. On the other hand, by doing this ordering check earlier, we save unnecessary computation, in case the update is not permitted because unsorted data

Reply:
Under this change, if the user passes a non-Timeseries argument to `update`, s/he will get an irrelevant error message about sorting instead of the correct error message that the argument type must be Timeseries indexed.

Reply:
move 299 above the new check